perma_web/api/views.py

```diff
@@ -279,11 +280,19 @@ class LinkFilter(django_filters.rest_framework.FilterSet):
     date = django_filters.IsoDateTimeFilter(field_name="creation_timestamp", lookup_expr='date')      # ?date=
     min_date = django_filters.IsoDateTimeFilter(field_name="creation_timestamp", lookup_expr='gte')   # ?min_date=
     max_date = django_filters.IsoDateTimeFilter(field_name="creation_timestamp", lookup_expr='lte')   # ?max_date=
-    url = django_filters.CharFilter(field_name="submitted_url", lookup_expr='icontains')              # ?url=
+    url = django_filters.CharFilter(method='surt_filter')                                             # ?url=
+
     class Meta:
         model = Link
         fields = ['url', 'date', 'min_date', 'max_date']
 
+    def surt_filter(self, queryset, name, value):
+        try:
+            canonicalized = surt.surt(value)
+        except ValueError:
+            return queryset
```

Comment:
What are the conditions that might throw a `ValueError` here? Would it be dangerous/non-secure in some way to return an unfiltered queryset? Should it at least log a warning?

Reply:
Ah, thanks, I will add a comment. `ValueError` is thrown when you pass something that can't be parsed into a surt, say, "sajhkdfkjasdhfjadkhfhas". I think it's perfectly fine to return a non-filtered queryset here: it would be just like they visited /api/v1/archives instead of /api/v1/archives?url=asljdhfladskjfhaldshf